### ISSUE TITLE
Addon functionality broken with new patch

### ISSUE THREAD

**Issue by Kylista** [AdiAddons/AdiBags#332]
Bag no longer opens and generates chat messages:

Error in AdiBags: Interface\Addons\AdiBags\core\OO.lui:54 CreateFrame(): Couldn't find inherited node "ItemButtonTemplate" -- details: [Error 00000177C390C300]
Error in Adibags: Interface\AddOns\AdiBags\core\Bags.lua:130: attempt to index field 'Closebutton' (a nil value) -- details: [Error 00000177C390C4E0]

**Comment by obliver27** [comment]
Can confirm it's happening to me and one guildie.

**Comment by Fizgig1973** [comment]
Happening as well. Only Adibags enabled, can only access Bank and Guild Bank, personal bags will not open.

**Comment by nkarasch** [comment]
[https://www.wowinterface.com/forums/showthread.php?p=331510](url)

> In 8.1.5 they changed how the item button template works. Instead of being a virtual button that is inherited it is now a new widget type that can be created directly.

> Old Way

> local button = CreateFrame("Button", "ButtonNameHere"..id, ParentFrame, "ItemButtonTemplate");

> New Way

> local button = CreateFrame("ItemButton", "ButtonNameHere"..id, ParentFrame, "OptionalTemplateHere");

I know nothing about lua or AdiBags structure, that is all I found about the changes so far

**Comment by Manabender** [comment]
I've made partial progress on a fix.

In AdiBags\core\OO.lua, replace line 54...

`local self = CreateFrame(class.frameType, addonName..class.name..class.serial, defaultParent, class.frameTemplate)`

...with these six lines...

`local self
if (class.frameTemplate == "ItemButtonTemplate") then
	self = CreateFrame("ItemButton", addonName..class.name..class.serial, defaultParent)
else
	self = CreateFrame(class.frameType, addonName..class.name..class.serial, defaultParent, class.frameTemplate)
end	`

The bag frame opens, and all the item buttons are present, but invisible. It works insofar as you can hover over them to see what they are, and click on them to move/use them, but you can't see their icons.

**Comment by Manabender** [comment]
Fix finished! Credit goes to semlar for helping with the other half of the fix.

In AdiBags\widgets\ItemButton.lua, replace line 61...

`local buttonClass, buttonProto = addon:NewClass("ItemButton", "Button", "ContainerFrameItemButtonTemplate", "ABEvent-1.0")`

...with...

`local buttonClass, buttonProto = addon:NewClass("ItemButton", "ItemButton", "ContainerFrameItemButtonTemplate", "ABEvent-1.0")`

**Comment by davidholland** [comment]
I issued a PR fixing the issues, hopefully it will either be merged soon or the author will make their own fix.

**Comment by neilseven13** [comment]
Confirmed the fix provided worked for me on all characters / servers in combination with using Elvui.

**Comment by samuraininja93** [comment]
Patch 8.1.5
AdiBags v1.9.13

Error in AdiBags: Interface\addOns\AdiBags\widgets\ContainerFrame/lua:714:attempt to index field 'BagSlotPanel' (a nil value) -- details: [Error 000002042144FF10]

Error in AdiBags: Interface\addOns\AdiBags\widgets\ContainerFrame/lua:97:attempt to index field 'Anchor' (a nil value) -- details: [Error 0000020453680F80]

Error in AdiBags: Interface\addOns\AdiBags\widgets\ContainerFrame/lua:555:attempt to index field 'Title' (a nil value) -- details: [Error 0000020453681070]

**Comment by toxxi** [comment]
Applied Manabender/semlar's fixes but am now receiving the following error:

[22:52:13] Error in AdiBags: Interface\AddOns\AdiBags\core\Bags.lua:130: attempt to index field 'frame' (a nil value) -- details: [Error 00000036B78F0B20]

Anyone else getting this?

**Comment by MuTLY** [comment]
To anyone else not getting the fix to work, make sure that you have the RELEASE version of the addon before trying to modify the files. Full fix below, credit to the original authors @Manabender @semlar @nkarasch 

In AdiBags\core\OO.lua, replace line 54...

`local self = CreateFrame(class.frameType, addonName..class.name..class.serial, defaultParent, class.frameTemplate)`

...with these lines...

`local self`
`if (class.frameTemplate == "ItemButtonTemplate")`
`then self = CreateFrame("ItemButton", addonName..class.name..class.serial, defaultParent)`
`else self = CreateFrame(class.frameType, addonName..class.name..class.serial, defaultParent, class.frameTemplate)`
`end`

In AdiBags\widgets\ItemButton.lua, replace line 61...

`local buttonClass, buttonProto = addon:NewClass("ItemButton", "Button", "ContainerFrameItemButtonTemplate", "ABEvent-1.0")`

...with...

`local buttonClass, buttonProto = addon:NewClass("ItemButton", "ItemButton", "ContainerFrameItemButtonTemplate", "ABEvent-1.0")`

In AdiBags\widgets\BagSlots.lua, replace line 219...

`local bagButtonClass, bagButtonProto = addon:NewClass("BagSlotButton", "Button", "ItemButtonTemplate", "ABEvent-1.0")`

...with...

`local bagButtonClass, bagButtonProto = addon:NewClass("BagSlotButton", "ItemButton", nil, "ABEvent-1.0")`

**Comment by grimmtooth** [comment]
Release version here, above fixes applied and do appear to address all the baddies. Thanks!

**Comment by Bak3y** [comment]
Confirmed @MuTLY's changes work.

**Comment by sbm128** [comment]
Dang, not working on this end. :\

**Comment by FelixFTW** [comment]
Confirmed.  Manabender's recommendation working for me.

**Comment by TKardinal** [comment]
Also confirmed. Manabender's recommendation working for me.

**Comment by SpareSimian** [comment]
Suggested change to issue title: "Bags don't open in 8.1.5". (I think only Kylista can edit the title.)

Can anyone point to documentation or forum posts explaining what happened to the GUI system that required this? I'm familiar with GUIs in other systems but don't have my head wrapped around how WoW works.  nkarasch's link assumes some knowledge I don't have.

**Comment by SpareSimian** [comment]
I made the edits to v1.9.13 and am seeing errors similar to samuraininja93's. The first error I see is:

OO.lua:57: CreateFrame(): Couldn't find inherited node "nil"

**Comment by SpareSimian** [comment]
I threw a print into Create_Frame to see what has a nil class.frameTemplate and it reports Button AdiBagsAnchor1.

**Comment by tflo** [comment]
Yep, the fix [as posted here](url
https://github.com/AdiAddons/AdiBags/issues/332#issuecomment-472213355) by @MuTLY seems to work fine with 1.9.13. Thanks.

**Comment by brooks986** [comment]
Cant thank you enough boys and girls , this worked 100% , thank you.  @MuTLY @Manabender @semlar @nkarasch

**Comment by toxxi** [comment]
This feels like a stupid question but which is the "release" version of AdiBags? I am reinstalling this version: https://www.wowace.com/projects/adibags

Are the fixes working with this version?

**Comment by unk1nd** [comment]
Confirmed working for me as well on version v1.9.13 from twitch client and after applying the code lines in the files.

**Comment by tflo** [comment]
> This feels like a stupid question but which is the "release" version of AdiBags? I am reinstalling this version: https://www.wowace.com/projects/adibags
> 
> Are the fixes working with this version?

In the side bar (right-hand side, under “Recent Files”) of that wowace page you find a download link labeled as “v1.9.13”. I assume this is the same 1.9.13 version as the one via Twitch (which I am using), so I would say it should work.

In case of doubt, open the the “AdiBags.toc” file of the downloaded plugin with a text editor, go to line 25, and it should say “v1.9.13” there.

**Comment by wayneluke** [comment]
Thanks for the code changes. I was able to get the addon to work again.

**Comment by Manabender** [comment]
So...interesting note. Turns out my fix (that is, my "half" of the fix) isn't actually necessary. AdiBags\core\OO.lua doesn't and didn't need to be changed at all. That change was just a basic scrub-tier hack which, while informative, was nothing but a bandaid to stop the bleeding. @semlar 's two changes tackled the root cause.

@nkarasch still helped. He didn't contribute any code, but he did save us (read: @semlar ) a lot of digging and head-scratching.

To summarize: OO.lua can be reverted by reversing the change I previously proposed, and the addon will still work.
